### PR TITLE
Do not copy initial recovery filter during split

### LINKED ISSUE #43955: index.routing.allocation.initial_recovery._id split after shrink
**Elasticsearch version** (`bin/elasticsearch --version`): 7.1.1

**Plugins installed**: []

**JVM version** (`java -version`):

**OS version** (`uname -a` if on a Unix-like system):

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

 1. Create an index
 2. Shrink an index
 3. remove the node the shrink occurred on from the cluster
 4. Wait for index to recover
 5. attempt to split the index

**Provide logs (if relevant)**:
Split fails on allocation. Decider says all primary shards must be on the dead node. Can not remove index.routing.allocation.initial_recovery._id setting from index.

### PR DESCRIPTION
**Pull request by DaveCTurner**
If an index is the result of a shrink then it will have a value set for
`index.routing.allocation.initial_recovery._id`. If this index is subsequently
split then this value will be copied over, forcing the initial allocation of
the split shards to occur on the node on which the shrink took place. Moreover
if this node no longer exists then the split will fail.  This commit suppresses
the copying of this setting when splitting an index.

Fixes #43955

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java [hunk 1]
```diff
@@ -554,4 +556,46 @@ public void testShrinkCommitsMergeOnIdle() throws Exception {
...
+    public void testShrinkThenSplitWithFailedNode() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(3);
+
+        final int shardCount = between(2, 5);
+        prepareCreate("original").setSettings(Settings.builder().put(indexSettings())
+            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, shardCount)).get();
+        for (int i = 0; i < 30; i++) {
+            client().prepareIndex("original", "_doc")
+                .setSource("{\"foo\" : \"bar\", \"i\" : " + i + "}", XContentType.JSON).get();
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java]: NIT: `.setSource("foo", "bar", "i" , i)` seems nicer? :)

**Review by original-brownbear — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java]: b1803fc459c is nicer still ;)

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/docbldesx (issue looks infrastructural, hopefully transient)

**Pull request merged**